### DISCUSSION TITLE
linear: vector must have 1 or 3 elements

### DISCUSSION THREAD

**Discussion by yanboma** [libvips/libvips#3357] [General]
when call  v_image.dzsave("/tmp/abc", VImage::option()->set("basename", "/tmp/test_abc")->set("overlap", 0)->set("tile_size", 256)->set("compression", -1)->set("background", to_vectorv(255))->set("skip_blanks", 0)->set("suffix", ".png"));

throw exception:
linear: vector must have 1 or 3 elements

**Comment by jcupitt** [comment]
Hello @yanboma,

You need to post a complete (small!) program that reproduces the problem. There's not enough information here for anyone to be able to say what's wrong.

**Comment by jcupitt** [comment]
Actually, it's this:

```C++
    ->set("background", to_vectorv(255))
```

The first parameter of vectorv is the size of the vector. Try:

```C++
    ->set("background", to_vectorv(1, 255))
    ->set("background", to_vectorv(3, 255, 255, 255))
    ->set("background", 255)
```

Any of those should work.

**Reply by jcupitt** [reply]
Or:

```C++
                ->set("background", new std::vector<int>({255, 255, 255}))
```

If you're lucky enough to be on c++11.

**Comment by yanboma** [comment]
got it，thanks
